sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -306,7 +347,7 @@ public RuntimeException logExceptionAsError(RuntimeException runtimeException) {
      * @param format formattable message.
      * @param args Arguments for the message, if an exception is being logged last argument is the throwable.
      */
-    private void performLogging(LogLevel logLevel, boolean isExceptionLogging, String format, Object... args) {
+    private void performLogging(LogLevel logLevel, boolean isExceptionLogging, boolean isSupplierLogging, String format, Object... args) {
```

Comment:
I'm wondering if there should be another implementation method, such as `performDeferredLogging`, which takes `Supplier<String>` and `Throwable` and the methods below could be removed.

Reply:
Sure, I'll implement it. 
Also, is it a possible scenario, someone added a supplier earlier into args? I have added this boolean for safety,  the isSupplierLogging method itself can determine wether is it supplier logging or not.

Reply:
@alzimmermsft Pushed a new version with performDeferredLogging. Some refactor was necessary to avoid major code duplication.